src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int VALUE_COUNT = 100;
+    private static final int FIGURE_COUNT = 5;
+    private static final int DEFAULT_VALUE = 10;
+    private final Random random = new Random();
+    private final ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(FIGURE_COUNT)) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomIsoscelesTrapezoid();
+            case 2:
+                return getRandomRectangle();
+            case 3:
+                return getRandomRightTriangle();
+            case 4:
+                return getRandomSquare();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    private Figure getRandomCircle() {
+        return new Circle(random.nextInt(VALUE_COUNT), color.getRandomColor());
+    }
+
+    private Figure getRandomIsoscelesTrapezoid() {
+        return new IsoscelesTrapezoid(random.nextInt(VALUE_COUNT),
+                random.nextInt(VALUE_COUNT),
+                random.nextInt(VALUE_COUNT),
+                color.getRandomColor());
+    }
+
+    private Figure getRandomRectangle() {
+        return new Rectangle(random.nextInt(VALUE_COUNT),
+                random.nextInt(VALUE_COUNT),
+                color.getRandomColor());
+    }
+
+    private Figure getRandomRightTriangle() {
+        return new RightTriangle(random.nextInt(VALUE_COUNT),
+                random.nextInt(VALUE_COUNT),
+                color.getRandomColor());
+    }
+
+    private Figure getRandomSquare() {
+        return new Square(random.nextInt(VALUE_COUNT), color.getRandomColor());
+    }
+
+    public Figure getDefaultFigure() {
```

Comment:
public methods should go before private methods

Reply:
in the condition of the task, he writes something else
![image](https://user-images.githubusercontent.com/98239700/211890920-87604a81-2e56-4c56-aa92-e2f02e168d6e.png)

Reply:
yes, this method has to be public, so it should appear before all private methods in the class, just change their order